zbuild.h

```diff
@@ -245,4 +245,10 @@
 #  endif
 #endif
 
+#ifndef Z_MEMORY_SANITIZER
+static inline void __msan_unpoison(const volatile void *a, size_t size)
```

Comment:
I think you can probably use a #define instead of empty function?

Reply:
Yes, but an empty function has a couple advantages:
- Variables passed to them are considered used by `-Wunused-variable` (this is relevant for the DFLTCC code)
- Type checking of arguments (problems introduced by non-MSan builds are found right away and not deferred to CI)

Reply:
The ``a`` and ``size`` parameters may generate an "unused variable" warning in the ``__msan_unpoison`` function.

Reply:
It's better to mark the function parameters "unused" explicitly...

Reply:
I don't like the idea of starting to include static functions in zbuild.h which is used everywhere.
Something like `#define _msan_posion(x, y) 1` is simple and will lead to less warnings.

Reply:
Would `#define _msan_posion(x, y) do { (void)(x); (void)(y); } while (0)` be okay? Otherwise `orig_t2` in `dfltcc()` will need to keep the `#ifdef` guard, making the things not as nice as they could've been.

Reply:
> I don't like the idea of starting to include static functions in zbuild.h which is used everywhere.
> Something like `#define _msan_posion(x, y) 1` is simple and will lead to less warnings.

`static` here merely means the function is local to current compilation unit... This allows it to be inlined on all common compilers and doesn't throw error due to duplicate symbols.

I'm not sure introducing dead code is the best practice, but I've seen other projects do the same for functions that are used only for some configurations.